app/controllers/avo/application_controller.rb

```diff
@@ -119,7 +119,22 @@ def detect_fields
     def set_related_resource
       raise Avo::MissingResourceError.new(related_resource_name) if related_resource.nil?
 
-      @related_resource = related_resource.new(params: params, view: action_name.to_sym, user: _current_user, record: @related_record).detect_fields
+      action_view = action_name.to_sym
+
+      # Get view from params unless actions is index or show or forms...
+      # Else, for example for detach action we want the view from params to can fetch the correct fields
+      view = if action_view.in?([:index, :show, :new, :edit, :create])
+        action_view
+      else
+        params[:view] || action_view
+      end
```

Comment:
This logic avoid the following scenario:

When a has many field is rendered the action is `index` and `params[:view]` is `show` or `edit` but we want to keep `@view` as `index` for the `related_resource` 

Same do not happen with other actions except the list on L126

Reply:
Please add this as a comment in the code so we know why it's there.